src/game_api/rpc.cpp

```diff
@@ -1868,3 +1868,139 @@ void set_start_level_paused(bool enable)
+    {
+        char* szDllName = (char*)PtrFromRva(pDosHeader, pImportDescriptor[uIndex].Name);
+
+        // Is this our module?
+        if (_strcmpi(szDllName, szModuleName) != 0)
+            continue;
+
+        if (!pImportDescriptor[uIndex].FirstThunk || !pImportDescriptor[uIndex].OriginalFirstThunk)
+            return FALSE;
+
+        PIMAGE_THUNK_DATA pThunk = (PIMAGE_THUNK_DATA)PtrFromRva(pDosHeader, pImportDescriptor[uIndex].FirstThunk);
+        PIMAGE_THUNK_DATA pOrigThunk = (PIMAGE_THUNK_DATA)PtrFromRva(pDosHeader, pImportDescriptor[uIndex].OriginalFirstThunk);
+
+        for (; pOrigThunk->u1.Function != NULL; pOrigThunk++, pThunk++)
+        {
+            // We can't process ordinal imports just named
+            if (pOrigThunk->u1.Ordinal & IMAGE_ORDINAL_FLAG)
+                continue;
+
+            PIMAGE_IMPORT_BY_NAME import = (PIMAGE_IMPORT_BY_NAME)PtrFromRva(pDosHeader, pOrigThunk->u1.AddressOfData);
+
+            // Is this our function?
+            if (_strcmpi(szFuncName, (char*)import->Name) != 0)
+                continue;
+
+            DWORD dwJunk = 0;
+            MEMORY_BASIC_INFORMATION mbi;
+
+            // Make the memory section writable
+            VirtualQuery(pThunk, &mbi, sizeof(MEMORY_BASIC_INFORMATION));
+            if (!VirtualProtect(mbi.BaseAddress, mbi.RegionSize, PAGE_EXECUTE_READWRITE, &mbi.Protect))
+                return FALSE;
+
+            // Save the old pointer
+            *pOldFunc = (PVOID*)(DWORD_PTR)pThunk->u1.Function;
+
+// Write the new pointer based on CPU type
+#ifdef _WIN64
+            pThunk->u1.Function = (ULONGLONG)(DWORD_PTR)pNewFunc;
+#else
+            pThunk->u1.Function = (DWORD)(DWORD_PTR)pNewFunc;
+#endif
+
+            if (VirtualProtect(mbi.BaseAddress, mbi.RegionSize, mbi.Protect, &dwJunk))
+                return TRUE;
+        }
+    }
+    return FALSE;
+}
+
+bool __stdcall QueryPerformanceCounterHook(LARGE_INTEGER* counter)
+{
+    QueryPerformanceCounter(&g_speedhack_current);
+    g_speedhack_fake.QuadPart += (long long)((g_speedhack_current.QuadPart - g_speedhack_prev.QuadPart) * g_speedhack_multiplier);
+    g_speedhack_prev = g_speedhack_current;
+    *counter = g_speedhack_fake;
+    return true;
+}
+
+void set_speedhack(std::optional<float> multiplier)
+{
+    g_speedhack_multiplier = multiplier.value_or(1.0f);
+    if (!g_speedhack_hooked)
+    {
+        QueryPerformanceCounter(&g_speedhack_prev);
+        g_speedhack_fake = g_speedhack_prev;
+        HookIAT("kernel32.dll", "QueryPerformanceCounter", QueryPerformanceCounterHook, &g_oldqpc);
+        g_speedhack_hooked = true;
+    }
+}
+
+float get_speedhack()
+{
+    return g_speedhack_multiplier;
+}
+
+void init_adventure()
+{
+    // TODO: I didn't check exactly what this does, but it fixes issues with character select being broken after quick start
+    static size_t offset = 0;
+    if (offset == 0)
+    {
+        offset = get_address("init_adventure");
+    }

```

Comment:
Why not just  `static size_t offset = get_address("init_adventure");` ?

Reply:
I dunno, we have this old crap laying around all over the place and I'm not really thinking when I copy paste it and change the pattern